python/shotgun_fields/bubble_widget.py

```diff
@@ -0,0 +1,495 @@
+        self._data = None
+
+        # should not grow or shrink
+        self.setSizePolicy(QtGui.QSizePolicy.Fixed, QtGui.QSizePolicy.Fixed)
+
+        # style the look of the bubble
+        self.setObjectName("bubble")
+        self.setStyleSheet(
+            """
+            #bubble {
+                border: 1px solid black;
+                border-radius: 5px;
+                background-color: %s;
+            } """ % self.palette().color(QtGui.QPalette.Button).name()
+        )
+
+
+        # create a remove button for the widget.
+        # extract a close button icon from the style and use it
+        self.remove_button = QtGui.QPushButton(self)
+        style = self.remove_button.style()
+        icon = style.standardIcon(style.SP_TitleBarCloseButton)
+        self.remove_button.setIcon(icon)
+        self.remove_button.setFlat(True)
+        self.remove_button.setStyleSheet("border: none")
+
+        # placeholder for the bubble's image
+        self.image_label = QtGui.QLabel(self)
+
+        # color the text to use the SG highlight color
+        text_color = QtGui.QColor(
+            sgtk.constants.SG_STYLESHEET_CONSTANTS["SG_HIGHLIGHT_COLOR"]
+        )
+        self.text_label = QtGui.QLabel(self)
+        palette = self.text_label.palette()
+        palette.setColor(QtGui.QPalette.WindowText, text_color)
+        self.text_label.setPalette(palette)
+
+        # layout the widgets
+        self.layout = QtGui.QHBoxLayout()
+        self.layout.setSizeConstraint(QtGui.QLayout.SetMinimumSize)
+        self.layout.setContentsMargins(3, 1, 3, 1)
+        self.layout.setSpacing(2)
+
+        self.layout.addWidget(self.image_label, QtCore.Qt.AlignVCenter)
+        self.layout.addWidget(self.text_label, QtCore.Qt.AlignVCenter)
+        self.layout.addWidget(self.remove_button, QtCore.Qt.AlignVCenter)
+        self.setLayout(self.layout)
+
+        # emit the "remove_clicked" signal when the button is clicked.
+        self.remove_button.clicked.connect(
+            lambda: self.remove_clicked.emit()
+        )
+
+    def set_text(self, label_text):
+        """
+        Set the bubble's display text.
+
+        :param str label_text: The display text
+        """
+        self.text_label.setText(label_text)
+
+    def set_image(self, url):
+        """
+        Set the bubble's display image.
+
+        :param str url:  The image url to display in the bubble.
+        """
+        self.image_label.setText("<img src='%s'/>" % url)
+
+    def set_removable(self, removable):
+        """
+        Set whether or not the bubble is removable.
+
+        Shows or hides the ``x`` button depending on the value of the
+        ``removable`` argument.
+        :param bool removable: ``True`` if the bubble is removable, ``False`` otherwise.
+        """
+        if removable:
+            self.remove_button.show()
+        else:
+            self.remove_button.hide()
+
+    def get_data(self):
```

Comment:
The idea here was to be a little closer to how Qt handles get/set of data for widgets. This also aligns with the fields widgets interface of get/set value. Feel free to add to hot topics if you think we should take a different approach.